fdbclient/SystemData.cpp

```diff
@@ -868,8 +868,32 @@ const KeyRef configKeysPrefix = configKeys.begin;
 
 const KeyRef perpetualStorageWiggleKey("\xff/conf/perpetual_storage_wiggle"_sr);
 const KeyRef perpetualStorageWiggleLocalityKey("\xff/conf/perpetual_storage_wiggle_locality"_sr);
-const KeyRef perpetualStorageWiggleIDPrefix("\xff/storageWiggleID/"_sr); // withSuffix /primary or /remote
+// The below two are there for compatible upgrade and downgrade. After 7.3, the perpetual wiggle related keys should use
+// format "\xff/storageWiggle/[primary | remote]/[fieldName]"
+const KeyRef perpetualStorageWiggleIDPrefix("\xff/storageWiggleID/"_sr); // withSuffix /primary/ or /remote/
 const KeyRef perpetualStorageWiggleStatsPrefix("\xff/storageWiggleStats/"_sr); // withSuffix /primary or /remote
+// "\xff/storageWiggle/[primary | remote]/[fieldName]"
+const KeyRef perpetualStorageWigglePrefix("\xff/storageWiggle/"_sr);
+
+// the final char is "/"
+const Key perpetualStorageWigglePrefixFor(bool primaryDc, PerpetualWigglePrefixType type) {
```

Comment:
A better way to handle state like this where there are different variables stored under a prefix would be following the model of `BackupConfig` and `RestoreConfig` where you have a struct that is given a base prefix and then there are named members which return KeyBackedTypes for sub-prefixes under that.   It allows defining FDB-backed structures with less code.